segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperServiceRunner.java

```diff
@@ -253,10 +258,37 @@ private void initializeZookeeper() throws Exception {
     private void runBookies() throws Exception {
         log.info("Starting Bookie(s) ...");
         for (int bkPort : this.bookiePorts) {
-            this.servers.add(runBookie(bkPort));
+            this.servers.add(runBookieWithRetry(bkPort));
         }
     }
 
+    private BookieServerAndResources runBookieWithRetry(int bkPort) throws Exception {
+        int count = 0;
+        boolean retry = true;
+        BookieServerAndResources bookieServerAndResources = null;
+        while (count < retries && retry) {
+            try {
+                log.info("Starting bookie with retry value of :" + count);
+                bookieServerAndResources = runBookie(bkPort);
+                retry = false;
+            } catch (BindException e) {
+                if ( ++count >= retries ) {
+                    e.printStackTrace();
+                    throw new BookieException(1) {
+                        @Override
+                        public String getMessage(int code) {
+                            return super.getMessage(code);
+                        }
+                    };
```

Comment:
Can we directly throw BindException from here, Instead of converting it to BookieException?

Reply:
yes, will be doing the same.